README.md

```diff
@@ -1,14 +1,49 @@
 # The Tracing Simulator
 
-## Notes
-If you want to make a pdf of the graph you are making for debugging purposes, install graphviz (https://graphviz.org/download/)
-and run with command line option"-g", eg, "cargo +nightly run -- -g".  Once you have run it, your graph will be in graph.pdf
+Tracing Simulator is a simulator for WebAssembly extensions.  With it you can define your own microservices architecture, 
+and run user-defined extensions on each service.
 
-## Build
-cargo +nightly build
+## Quickstart
+
+### Build
+```cargo +nightly build```
+
+### Run
+Without a filter:
+```cargo +nightly run```
+
+With a filter:
+```cargo +nightly run -- -p target/debug/libfilter_example.dylib```
+
+## Customization
+
+### Making your own microservices architecture
+
+### Making your own extension
+The recommended way to make an extension is by using the tracing compiler found [here](https://github.com/dyn-tracing/tracing_compiler) with the *-c sim* option.
+Then run:
+```cp tracing_compiler/rust_filter/filter.rs tracing_sim/libs/filter_lib/src/filter.rs```
```

Comment:
That workflow seems out of date. And it is also incorrect. It actually is 
```cp -R tracing_compiler/rust_filter tracing_sim/libs/rust_filter
cd tracing_sim/libs/rust_filter/
cargo +nightly build
cd ../..
cargo +nightly run -- -p libs/rust_filter/target/debug/librust_filter```

A shorthand of this is the command `./check_filter.py -qf count.cql -qu count` which can be found here https://github.com/dyn-tracing/tracing_env/blob/master/check_filter.py 
That file should probably be called `check_sim_filter`.